core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -97,4 +108,35 @@ private ProbeResult fromSCMLink(ProbeContext context, String scm) {
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    public static String searchPomFiles(File directory, String pluginName) {
+        File[] files = directory.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    searchPomFiles(file, pluginName);
+                } else {
+                    return getSCMFolderPath(file.getAbsolutePath(), pluginName);
+                }
+            }
+        }
+        return null;
+    }
+
+    public static String getSCMFolderPath(String filePath, String pluginName) {
+        if (filePath.startsWith("pom.xml")) {
+            MavenXpp3Reader mavenReader = new MavenXpp3Reader();
+            try (Reader reader = new InputStreamReader(new FileInputStream(filePath), StandardCharsets.UTF_8)) {
+                Model model = mavenReader.read(reader);
+                if (model.getPackaging().equals("hpi") && model.getArtifactId().equals(pluginName)) {
+                    return filePath + "/pom.xml";
+                }
+            } catch (IOException e) {
+                LOGGER.error("Pom file not found for {}", pluginName);
```

Comment:
Having the pass in the exception message could help debugging an issue.

Reply:
The reason the `HasUnreleasedProductionChangesProbeTest#shouldFailIfThereIsNotReleasedCommitsInModule` is failing because I am not passing the correct folder path.